src/NotInstantiableWithoutWidgetFactoryInitializationException.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Yiisoft\Widget;
+
+use Throwable;
+use Yiisoft\Definitions\Exception\NotInstantiableException;
+use Yiisoft\FriendlyException\FriendlyExceptionInterface;
+
+final class NotInstantiableWithoutWidgetFactoryInitializationException extends NotInstantiableException implements FriendlyExceptionInterface
+{
+    public function __construct(private Throwable $previous)
+    {
+        parent::__construct(
+            $previous->getMessage() .
+            ' Perhaps need initialize "WidgetFactory" with container for resolve dependencies.',
+            previous: $previous,
+        );
+    }
+
+    public function getName(): string
+    {
+        return 'Failed to create a widget. ' . $this->previous->getMessage();
+    }
+
+    public function getSolution(): ?string
+    {
+        return <<<'SOLUTION'
+            Perhaps need initialize `WidgetFactory` with container for resolve dependencies.
+
+            To initialize the widget factory call `WidgetFactory::initialize()` before using the widget.
+            It is a good idea to do that for the whole application.
```

Comment:
Please specify the file where the snippet can be placed

Reply:
This case for non-Yii environment. I don't know such file...